librz/core/cbin.c

```diff
@@ -327,14 +323,64 @@ RZ_API int rz_core_bin_set_by_name(RzCore *core, const char *name) {
 	return false;
 }
 
-RZ_API int rz_core_bin_apply_all_info(RzCore *r, RzBinFile *binfile) {
+RZ_API bool rz_core_bin_apply_info(RzCore *r, RzBinFile *binfile, int mask) {
+	rz_return_val_if_fail(r && binfile, false);
+	RzBinObject *binobj = binfile->o;
+	RzBinInfo *info = binobj ? binobj->info : NULL;
+	if (!info) {
+		return false;
+	}
+
+	// ----
+	// inlined: rz_core_bin_info(r, RZ_CORE_BIN_ACC_ALL, NULL, RZ_MODE_SET, va, NULL, NULL);
+
+	bool va = info->has_va;
+
+	if (mask & RZ_CORE_BIN_ACC_STRINGS) {
+		rz_core_bin_apply_strings(r, binfile);
+	}
+	if (mask & RZ_CORE_BIN_ACC_INFO) {
+		rz_core_bin_apply_config(r, binfile);
+	}
+	if (mask & RZ_CORE_BIN_ACC_MAIN) {
+		rz_core_bin_apply_main(r, binfile, va);
+	}
+	if (mask & RZ_CORE_BIN_ACC_DWARF) {
+		rz_core_bin_apply_dwarf(r, binfile);
+	}
+	if (mask & RZ_CORE_BIN_ACC_ENTRIES) {
+		rz_core_bin_apply_entry(r, binfile, va);
+	}
+	if (mask & RZ_CORE_BIN_ACC_SECTIONS) {
+		rz_core_bin_apply_sections(r, binfile, va);
+	}
+	if (mask & RZ_CORE_BIN_ACC_RELOCS && rz_config_get_i(r->config, "bin.relocs")) {
+		rz_core_bin_apply_relocs(r, binfile, va);
+	}
+	if (mask & RZ_CORE_BIN_ACC_IMPORTS) {
+		rz_core_bin_apply_imports(r, binfile, va);
+	}
+	if (mask & RZ_CORE_BIN_ACC_SYMBOLS) {
+		rz_core_bin_apply_symbols(r, binfile, va);
+	}
+	if (mask & RZ_CORE_BIN_ACC_CLASSES) {
+		rz_core_bin_apply_classes(r, binfile);
+	}
+	if (mask & RZ_CORE_BIN_ACC_RESOURCES) {
+		rz_core_bin_apply_resources(r, binfile);
+	}
+	// ----
+
+	return true;
```

Comment:
should we have a return value for failure in any of those methods called?

Reply:
Imo, a failure in one of these sub-functions should not really mean a failure of the whole function since other info would still be loaded successfully.

Reply:
agreed